patches/server/0034-Do-not-proceed-hopper-which-has-solid-block-above.patch

```diff
@@ -0,0 +1,48 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: jeffry1829 <jeffry1829@gmail.com>
+Date: Fri, 24 Apr 2020 00:41:42 +0800
+Subject: [PATCH] Do not proceed hopper which has solid block above
+
+
+diff --git a/src/main/java/net/minecraft/server/TileEntityHopper.java b/src/main/java/net/minecraft/server/TileEntityHopper.java
+index 4afd33a4916788305ab2e5810db00936a582c95f..d93c560556cf3a8584f27e419b937727bf542680 100644
+--- a/src/main/java/net/minecraft/server/TileEntityHopper.java
++++ b/src/main/java/net/minecraft/server/TileEntityHopper.java
+@@ -393,8 +393,36 @@ public class TileEntityHopper extends TileEntityLootable implements IHopper, ITi
+             return iinventory.getItem(i).isEmpty();
+         });
+     }
+-
++    // Tuinity - start Don't proceed hopper which has solid block above
++    public static boolean dontSuckByHopperBelow(Block b){ //newly defined
++        if(b==Blocks.DIRT||b==Blocks.STONE||b==Blocks.COBBLESTONE||b==Blocks.ANDESITE||b==Blocks.GRANITE||
```

Comment:
there has to be a better way than a thousand conditional checks.

Reply:
If other method didn't work well, maybe add a HashMap?
ez O(1) (?